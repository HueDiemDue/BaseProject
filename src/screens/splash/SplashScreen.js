import React from 'react'
import {
  View,
  StyleSheet,
  Text,
} from 'react-native'
import { setRootHomeScreen } from '../../navigator'
import { useNavigationAppearEvents } from '../../hooks'
import { LoadingIndicator } from '../../components'
import Colors from '../../styles/colors'

const SplashScreen = props => {

  const { componentId } = props

  useNavigationAppearEvents(componentId, {
    componentDidAppear: () => {
      setTimeout(() => {
        setRootHomeScreen()
      }, 2000)
    },
    componentDidDisappear: () => {

    }
  })

  return (
    <View style={styles.container}>
      <View style={styles.body}>
        <LoadingIndicator
          size={'large'}
        />
        <Text style={styles.text}>Hello</Text>
      </View>
    </View>
  )
}

export default SplashScreen

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  body: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Colors.blue_1
  },
  text: {
    fontSize: 14,
    fontWeight: 'bold',
    color: Colors.black
  }
})
