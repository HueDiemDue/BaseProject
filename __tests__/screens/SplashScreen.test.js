import React from 'react'
import { shallow, mount } from 'enzyme'
import SplashScreen from '../../src/screens/splash/SplashScreen'

describe('Screens > SplashScreen', () => {
  describe('render', () => {
    test('Render a snapshot for SplashScreen use renderer', () => {

      const wrapper = mount(<SplashScreen />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  // describe('componentDidMount', () => {
  //   let wrapper

  //   beforeEach(() => {
  //     wrapper = mount(<SplashScreen />)
  //   })
  //   test('should call componentDidMount lifecycle method', () => {
  //     // const instance = wrapper.instance()
  //     // // instance.componentDidMount = jest.fn()
  //     // const spy = jest.spyOn(instance, 'directScreen')
  //     // wrapper.instance().forceUpdate()
  //     // // wrapper.update()
  //     // // instance.componentDidMount()

  //     // expect(spy).toHaveBeenCalledTimes(1)

  //   })
  // })
})

// afterEach(() => {
//   spy.mockClear()
// })
