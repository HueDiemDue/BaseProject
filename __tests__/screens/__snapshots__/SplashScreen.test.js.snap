// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screens > SplashScreen render Render a snapshot for SplashScreen use renderer 1`] = `
<SplashScreen>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <react-native-mock>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(92,189,211,255)",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <react-native-mock>
          <LoadingIndicator
            size="large"
          >
            <ActivityIndicator
              color={null}
              size="large"
              style={
                Object {
                  "margin": 20,
                }
              }
            >
              <react-native-mock />
            </ActivityIndicator>
          </LoadingIndicator>
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            <react-native-mock>
              Hello
            </react-native-mock>
          </Text>
        </react-native-mock>
      </View>
    </react-native-mock>
  </View>
</SplashScreen>
`;
